openmetadata-service/src/main/java/org/openmetadata/service/apps/bundles/insights/workflows/costAnalysis/CostAnalysisWorkflow.java

```diff
@@ -0,0 +1,361 @@
+import org.openmetadata.schema.entity.services.DatabaseService;
+import org.openmetadata.schema.system.StepStats;
+import org.openmetadata.schema.type.LifeCycle;
+import org.openmetadata.service.Entity;
+import org.openmetadata.service.apps.bundles.insights.DataInsightsApp;
+import org.openmetadata.service.apps.bundles.insights.processors.CreateReportDataProcessor;
+import org.openmetadata.service.apps.bundles.insights.sinks.ReportDataSink;
+import org.openmetadata.service.apps.bundles.insights.utils.TimestampUtils;
+import org.openmetadata.service.apps.bundles.insights.workflows.WorkflowStats;
+import org.openmetadata.service.apps.bundles.insights.workflows.costAnalysis.processors.AggregatedCostAnalysisReportDataAggregator;
+import org.openmetadata.service.apps.bundles.insights.workflows.costAnalysis.processors.AggregatedCostAnalysisReportDataProcessor;
+import org.openmetadata.service.apps.bundles.insights.workflows.costAnalysis.processors.DatabaseServiceTablesProcessor;
+import org.openmetadata.service.apps.bundles.insights.workflows.costAnalysis.processors.RawCostAnalysisReportDataProcessor;
+import org.openmetadata.service.exception.SearchIndexException;
+import org.openmetadata.service.jdbi3.ListFilter;
+import org.openmetadata.service.jdbi3.ReportDataRepository;
+import org.openmetadata.service.jdbi3.TableRepository;
+import org.openmetadata.service.util.JsonUtils;
+import org.openmetadata.service.util.ResultList;
+import org.openmetadata.service.workflows.searchIndex.PaginatedEntitiesSource;
+
+@Slf4j
+public class CostAnalysisWorkflow {
+  public static final String AGGREGATED_COST_ANALYSIS_DATA_MAP_KEY =
+      "aggregatedCostAnalysisDataMap";
+  @Getter private final int batchSize;
+  @Getter private final Long startTimestamp;
+  @Getter private final Long endTimestamp;
+  private final int retentionDays = 30;
+  @Getter private final List<PaginatedEntitiesSource> sources = new ArrayList<>();
+
+  public record CostAnalysisTableData(
+      Table table, Optional<LifeCycle> oLifeCycle, Optional<Double> oSize) {}
+  ;
+
+  public record AggregatedCostAnalysisData(
+      Double totalSize,
+      Double totalCount,
+      DataAssetMetrics unusedDataAssets,
+      DataAssetMetrics frequentlyUsedDataAssets) {}
+  ;
+
+  @Getter private DatabaseServiceTablesProcessor databaseServiceTablesProcessor;
+  @Getter private RawCostAnalysisReportDataProcessor rawCostAnalysisReportDataProcessor;
+
+  @Getter
+  private AggregatedCostAnalysisReportDataProcessor aggregatedCostAnalysisReportDataProcessor;
+
+  @Getter private final WorkflowStats workflowStats = new WorkflowStats();
+
+  public CostAnalysisWorkflow(
+      Long timestamp, int batchSize, Optional<DataInsightsApp.Backfill> backfill) {
+    this.endTimestamp =
+        TimestampUtils.getEndOfDayTimestamp(TimestampUtils.subtractDays(timestamp, 1));
+    this.startTimestamp = TimestampUtils.getStartOfDayTimestamp(endTimestamp);
+
+    // TODO: Implement Backfill by using DataAsset Version.
```

Comment:
are we tracking these TODOs in a ticket?

Reply:
They are not, I'll track it in the Data Insights Design document at least!